provider/permissions_data_test.go

```diff
@@ -0,0 +1,68 @@
+package provider
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccPermissionsDataSource_basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testFireHydrantIsSetup(t) },
+		ProviderFactories: defaultProviderFactories(),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccPermissionsDataSourceConfig(),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttrSet("data.firehydrant_permissions.all", "id"),
+					resource.TestCheckResourceAttrSet("data.firehydrant_permissions.all", "permissions.#"),
+					// Check that we get permissions and they have the expected structure
+					resource.TestCheckResourceAttrSet("data.firehydrant_permissions.all", "permissions.0.slug"),
+					resource.TestCheckResourceAttrSet("data.firehydrant_permissions.all", "permissions.0.display_name"),
+					resource.TestCheckResourceAttrSet("data.firehydrant_permissions.all", "permissions.0.description"),
+					resource.TestCheckResourceAttrSet("data.firehydrant_permissions.all", "permissions.0.category_slug"),
+					resource.TestCheckResourceAttrSet("data.firehydrant_permissions.all", "permissions.0.category_display_name"),
+					resource.TestCheckResourceAttrSet("data.firehydrant_permissions.all", "permissions.0.available"),
+				),
+			},
+		},
+	})
+}
+
+// Test that demonstrates the workflow: get permissions, then create a role with them
+func TestAccPermissionsAndRoleWorkflow(t *testing.T) {
```

Comment:
This isn't actually using the data source, so it becomes just another test that could potentially break if the role resource isn't working correctly.  I'd remove this and `testAccPermissionsAndRoleWorkflowConfig()` that it uses for simplicity's sake and keep all of the role stuff isolated in its own tests.